src/indexed_crate.rs

```diff
@@ -8,15 +8,23 @@ pub struct IndexedCrate<'a> {
 
     // For an Id, give the list of item Ids under which it is publicly visible.
     pub(crate) visibility_forest: HashMap<&'a Id, Vec<&'a Id>>,
+
+    /// As of https://github.com/rust-lang/rust/pull/105182
+    /// the `rustdoc_types::Trait`s from foreign/outside crates
+    /// (in particular, traits like `Debug`, `Send` or `Eq`) are no longer present in `inner`.
+    /// Unfortunately, the current schema requires having `rustdoc_types::Trait` for each
+    /// trait implementation.
+    /// The `dummy_trait_items` tries to fix this problem by creating a hand-written `Trait`
+    /// for each of the std traits the user can use.
```

Comment:
To make this easier to understand, lead with the impact, then add the context. If someone sees this comment, we want them to be able to quickly assess whether they might be affected by it or not.

What do you think of this?
```suggestion
    /// Trait items defined in external crates are not present in the `inner: &Crate` field,
    /// even if they are implemented by a type in that crate. This also includes
    /// Rust's built-in traits like `Debug, Send, Eq` etc.
    ///
    /// This change is approximately as of rustdoc v23,
    /// in https://github.com/rust-lang/rust/pull/105182
    ///
    /// As a temporary workaround, we manually create the trait items
    /// for the most common Rust built-in traits and link to those items
    /// as if they were still part of the rustdoc JSON file.
    ///
    /// A more complete future solution may generate multiple crates' rustdoc JSON
    /// and link to the external crate's trait items as necessary.
```

Reply:
For extra bonus points: I noticed that `IndexedCrate` is a pub type without a doc comment, and it would be good to have something substantially similar to what I wrote here documented on the public type itself. The change we're making affects which traits are shown as implemented, so it would be good to have it documented prominently on the type itself rather than on a `pub(crate)` (i.e. private) field.

Reply:
Done in 0924a99007bedee1516fe6eae723072ba4d7ff5b.